backend/app/project/routes.py

```diff
@@ -1,8 +1,60 @@
-import uuid
 from flask import jsonify, request, abort
 from app import db
-from app.models import Project
 from . import project
+from app.models import Project
+
+
+# get all endpoints
+@project.route("", methods=["GET"])
+def get_projects():
+    projects = Project.query.all()
+    return jsonify(Project.serialize_list(projects))
+
+
+# create a new project
+@project.route("", methods=["POST"])
+def create_project():
+    data = request.get_json(force=True)
+    address = data.get("address")
+    city = data.get("city")
+    province = data.get("province")
+    postal_code = data.get("postal_code")
+    neighbourhood = data.get("neighbourhood")
+    year = data.get("year")
+    name = data.get("name")
+    type = data.get("type")
+    contacts = data.get("contacts")
+
+    # check if all fields are empty, if so it's a garbage post
+    if (
+        address == ""
+        and city == ""
+        and province == ""
+        and postal_code == ""
+        and neighbourhood == ""
+        and year is None
+        and name == ""
+        and type == ""
+        and len(contacts) == 0
+    ):
+        abort(400, "Cannot have all empty fields for a new project")
+
+    new_project = Project(
+        address=address,
+        city=city,
+        province=province,
+        postal_code=postal_code,
+        neighbourhood=neighbourhood,
+        year=year,
+        name=name,
+        type=type,
+        contacts=contacts,
```

Comment:
@peterghrong Sorry can't believe I missed this, but when we're grabbing contacts from the request payload, would we not want to initialize it into a Contact class before assigning it to the Project? I think this will lead to a runtime error.

Reply:
Good point, I think this will depend on the contact endpoints though